OpenRA.Mods.D2k/Activities/SwallowActor.cs

```diff
@@ -78,19 +77,8 @@ bool AttackTargets(Actor self, IEnumerable<Actor> targets)
 			var affectedPlayers = targets.Select(x => x.Owner).Distinct().ToList();
 			Game.Sound.Play(SoundType.World, swallow.Info.WormAttackSound, self.CenterPosition);
 
-			Game.RunAfterDelay(1000, () =>
-			{
-				if (!Game.IsCurrentWorld(self.World))
-					return;
-
-				foreach (var player in affectedPlayers)
-				{
-					Game.Sound.PlayNotification(player.World.Map.Rules, player, "Speech", swallow.Info.WormAttackNotification, player.Faction.InternalName);
-
-					if (player == player.World.RenderPlayer)
-						radarPings.Add(() => true, attackPosition, Color.Red, 50);
-				}
-			});
+			foreach (var player in affectedPlayers)
+				self.World.AddFrameEndTask(w => w.Add(new MapNotificationEffect(player, "Speech", swallow.Info.WormAttackNotification, 25, true, attackPosition, Color.Red, 50)));
```

Comment:
Isn't `50` already the default for `pingDuration`? (So you can drop it here and you could even make a default for the red color, while you're at it.) Not really a blocker though.